server/with-bun/handlers/signWithParaClient.ts

```diff
@@ -1,95 +1,127 @@
+    }
 
-  // Retrieve and decrypt key share
-  const keyShare = getKeyShareInDB(email);
-  if (!keyShare) {
-    return new Response("Key share does not exist", { status: 400 });
-  }
+    const keyShare = getKeyShareInDB(email);
+    if (!keyShare) {
+      return new Response(`Key share not found in DB for ${email}`, { status: 404 });
+    }
+
+    let decryptedKeyShare: string;
+    try {
+      decryptedKeyShare = await decrypt(keyShare);
+    } catch (decryptionError) {
+      console.error(`Failed to decrypt key share for ${email}:`, decryptionError);
+      return new Response("Failed to process key share", { status: 500 });
+    }
+
+    await para.setUserShare(decryptedKeyShare);
+
+    const wallets = await para.getWallets();
+    if (!wallets || Object.keys(wallets).length === 0) {
+      throw new Error("Failed to load wallet details after setting user share.");
+    }
 
-  const decryptedKeyShare = decrypt(keyShare);
-  await para.setUserShare(decryptedKeyShare);
-
-  // Get wallet details
-  const wallets = await para.getWallets();
-  const wallet = Object.values(wallets)[0];
-  const walletId = wallet.id;
-  const walletAddress = wallet.address;
-
-  // Prepare raw transaction data. These are dummy values and should be replaced with actual values.
-  const demoRawTx = {
-    nonce: "0x00", // Should match the current nonce on the network
-    gasPrice: "0x09184e72a000", // Should match the current gas price on the network
-    gasLimit: "0x2710", // Gas limit for the transaction
-    to: walletAddress, // Destination address
-    from: walletAddress, // Source address
-    value: "0x00", // No Ether transfer (0 value)
-    data: "0x", // No data being sent
-  };
-
-  // RLP encode the transaction and convert to base64 (before signing)
-  const rlpEncodedTx = RLP.encode([
-    demoRawTx.nonce,
-    demoRawTx.gasPrice,
-    demoRawTx.gasLimit,
-    demoRawTx.to,
-    demoRawTx.from,
-    demoRawTx.value,
-    demoRawTx.data,
-  ]);
-
-  // Convert the RLP encoded transaction to base64 for signing
-  const rlpEncodedTxBase64 = Buffer.from(rlpEncodedTx).toString("base64");
-
-  // Sign the transaction
-  const signTransactionResult = await para.signTransaction({ walletId, rlpEncodedTxBase64, chainId: "11155111" });
-
-  // Return the final signed transaction result
-  return new Response(JSON.stringify({ route: "signWithParaPreGen", signTransactionResult }), {
-    status: 200,
-  });
+    const wallet = Object.values(wallets)[0];
+    const walletId = wallet.id;
+    const walletAddress = wallet.address;
+
+    if (!walletId || !walletAddress) {
+      throw new Error("Failed to retrieve wallet ID or address.");
+    }
+
+    const chainId = "11155111";
+
+    const nonce = 0n;
+    const gasLimit = 21000n;
+    const gasPriceInGwei = 20n;
+    const valueInEth = 0n;
+
+    const gasPriceInWei = gasPriceInGwei * 10n ** 9n;
+    const valueInWei = valueInEth * 10n ** 18n;
+
+    const demoRawTxData = {
```

Comment:
do we have helper fn's for constructing demo transactions? I recall us adding this a while back